samples/cpp/prompt_lookup_decoding_lm/prompt_lookup_decoding_lm.cpp

```diff
@@ -47,13 +47,18 @@ struct TextStreamer {
         if (text.size() >= 3 && text.compare(text.size() - 3, 3, "�") == 0) {
             // Don't print incomplete text
             return;
+        } else if (text.size() > print_len && std::any_of(text.begin() + print_len, text.end(), [](char c) { return c == ' '; })) {
+            // It is possible to have a shorter text after adding new token.
+            // Print to output only if text lengh is increaesed and if containt full words.
```

Comment:
While you can't add it to tests (I guess because permission is need to download llama3), can you describe an example in code comments. For example, requirement for the increasing sequence is clear (although example would be nice anyway), but why is space required?

Reply:
I took already existing solution CVS-143742 As i understood we check for spaces to print only if there are complete words, but i think such strict check it's not necessary, not i removed it.

I don't face this error locally on the reported model and prompt (the code below run on master)
![image](https://github.com/openvinotoolkit/openvino.genai/assets/5703039/f76fc737-790b-4b16-b2d9-47e111034a55)

I was told that in this dataset 8 of 1000 prompts have this issue 
https://huggingface.co/datasets/anon8231489123/ShareGPT_Vicuna_unfiltered/blob/main/ShareGPT_V3_unfiltered_cleaned_split.json

I will cycle through prompts from there to reproduce issue and add example with problematic, and will add test at least to check it locally.

Reply:
I encourage you can ask why they used spaces in their fix

Reply:
i didn't ask yet, but i suspected they looked to HF
https://github.com/huggingface/transformers/blob/main/src/transformers/generation/streamers.py#L108-L112

Reply:
i found prompt on which issue is reproduced. This 
`'I have an interview about product speccing with the company Weekend Health. Give me an example of a question they might ask with regards about a new feature'` generates sequence which can become shorter when new tokens are added

![image](https://github.com/openvinotoolkit/openvino.genai/assets/5703039/23dcc9e6-e76a-44c3-a246-985796343c18)

Added test `test_callback_decoding_metallama` for that, but on precommit it's skipped. Also added unicode decoding test for tokenizer, but xfailed them until we enable ov::Tensor in pybind.